verilog/preprocessor/verilog_preprocess.cc

```diff
@@ -587,48 +598,19 @@ absl::Status VerilogPreprocess::HandleInclude(
   std::filesystem::path file_path =
       std::string(token_text.substr(1, token_text.size() - 2));
 
-  std::string file_path_to_search = file_path.string();
-
-  if (file_path.is_relative()) {
-    bool found = false;
-    for (const auto& base_dir : preprocess_info_.include_dirs) {
-      file_path_to_search =
-          verible::file::JoinPath(base_dir, file_path.string());
-      auto search_status = verible::file::FileExists(file_path_to_search);
-      if (search_status.ok()) {
-        found = true;
-        break;
-      }
-    }
-    if (!found) {
-      preprocess_data_.errors.push_back(
-          {**token_iter,
-           absl::StrCat(file_path_to_search, ": no such file found.")});
-      return absl::InvalidArgumentError("ERROR: included file can't be found.");
-    }
-  } else {
-    if (auto status = verible::file::FileExists(file_path_to_search);
-        !status.ok()) {
-      preprocess_data_.errors.push_back(
-          {**token_iter, std::string(status.message())});
-      return status;
-    }
-  }
-
-  // Actually open the file.
-  std::string source_contents;
-  if (auto status =
-          verible::file::GetContents(file_path_to_search, &source_contents);
-      !status.ok()) {
+  // Use the provided FileOpener to open the included file.
+  const auto status_or_file = file_opener_(file_path.string());
+  if (!status_or_file.ok()) {
     preprocess_data_.errors.push_back(
-        {**token_iter, std::string(status.message())});
-    return status;
+        {**token_iter, std::string(status_or_file.status().message())});
+    return status_or_file.status();
   }
+  const auto source_contents = *status_or_file;
```

Comment:
Instead of `auto`, I'd probably be specific here with the type, as it might be important for the reader to know that this is a `absl::string_view`.